src/test/java/com/google/cloud/genomics/dataflow/pipelines/CountReadsITCase.java

```diff
@@ -44,22 +53,63 @@ public void voidEnsureEnvVar() {
   public void testLocal() throws Exception {
     final String OUTPUT = TEST_OUTPUT_GCS_FOLDER + "CountReadsITCase-testLocal-output.txt";
     String[] ARGS = {
+        "--apiKey=" + API_KEY,
+        "--project=genomics-pipelines",
+        "--output=" + OUTPUT,
+        "--references=1:550000:560000",
+        "--BAMFilePath=" + TEST_BAM_FNAME
+    };
+    final long EXPECTED = 685;
+    GenomicsOptions popts = PipelineOptionsFactory.create().as(GenomicsOptions.class);
+    popts.setApiKey(API_KEY);
+    GcsUtil gcsUtil = new GcsUtil.GcsUtilFactory().create(popts);
+    touchOutput(gcsUtil, OUTPUT);
+
+    CountReads.main(ARGS);
+
+    BufferedReader reader = new BufferedReader(Channels.newReader(gcsUtil.open(GcsPath.fromUri(OUTPUT)), "UTF-8"));
+    long got = Long.parseLong(reader.readLine());
+
+    Assert.assertEquals(EXPECTED, got);
+  }
+
+  /**
+   * CountReads running on Dataflow.
+   */
+  @Test
+  public void testCloud() throws Exception {
+    final String OUTPUT = TEST_OUTPUT_GCS_FOLDER + "CountReadsITCase-testCloud-output.txt";
+    String[] ARGS = {
         "--apiKey="+API_KEY,
         "--project=genomics-pipelines",
         "--output=" + OUTPUT,
+        "--numWorkers=2",
+        "--runner=BlockingDataflowPipelineRunner",
+        "--stagingLocation=" + TEST_STAGING_GCS_FOLDER,
         "--references=1:550000:560000",
-        "--BAMFilePath="+TEST_BAM_FNAME
+        "--BAMFilePath=" + TEST_BAM_FNAME
     };
     final long EXPECTED = 685;
     GenomicsOptions popts = PipelineOptionsFactory.create().as(GenomicsOptions.class);
     popts.setApiKey(API_KEY);
     GcsUtil gcsUtil = new GcsUtil.GcsUtilFactory().create(popts);
+    touchOutput(gcsUtil, OUTPUT);
 
     CountReads.main(ARGS);
 
     BufferedReader reader = new BufferedReader(Channels.newReader(gcsUtil.open(GcsPath.fromUri(OUTPUT)), "UTF-8"));
     long got = Long.parseLong(reader.readLine());
+
     Assert.assertEquals(EXPECTED, got);
   }
 
+  /**
+   * make sure we can get to the output, and at the same time avoid a false negative if
```

Comment:
Yes. Deleting the outputs will have to be a separate checkin because it's
not trivial: the util classes don't expose a delete method (and I can't
just use gsutil)

On Fri, May 8, 2015 at 1:13 PM, Nicole Deflaux <notifications@github.com>
wrote:

> In
> src/test/java/com/google/cloud/genomics/dataflow/pipelines/CountReadsITCase.java
> <https://github.com/googlegenomics/dataflow-java/pull/71#discussion_r29972579>
> :
>
> >      Assert.assertEquals(EXPECTED, got);
> >    }
> >
> > +  /**
> > +   * make sure we can get to the output, and at the same time avoid a false negative if
>
> With the failsafe plugin you can have a code hook that runs before all the
> tests (e.g. make a temp folder in GCS for all test output) and a code hook
> that runs after all the integration tests regardless of whether they pass
> or fail (e.g., rm -r the temp GCS folder).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/googlegenomics/dataflow-java/pull/71/files#r29972579>.
>